tests/integration_tests/allennlp_tests/test_allennlp.py

```diff
@@ -8,7 +8,7 @@
 import optuna
 
 
-def test__set_param() -> None:
+def test__build_params() -> None:
```

Comment:
A nit.

```suggestion
def test_build_params() -> None:
```

Reply:
Thank you for the comment.
I named the test `test__build_params` because these are the tests for private methods (`test_` + `_build_params`).
https://github.com/optuna/optuna/pull/1449/files#diff-194650369fb014d2ea8f428774c04276L102

Do you think it's okay to remove consecutive underscores?

Reply:
Hmm, we use both `test_` and `test` for the test functions of the private methods.

We have three test functions which use `test_` as prefix:
- `tests/integration_tests/allennlp_tests/test_allennlp.py:11:def test__set_param() -> None:`
- `tests/integration_tests/lightgbm_tuner_tests/test_alias.py:5:def test__handling_alias_parameters():`
- `tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:337:    def test__parse_args_wrapper_args(self):`

Most of the test functions employ `test`, although `test_` seems to follow the naming convention.
@keisuke-umezawa @hvy Do you have any ideas about it?

Reply:
Hm, I think both makes sense but I'd also suggest removing the second underscore(s). A test function/method doesn't necessarily have to exactly match one particular function/method/etc. It could test multiple, for instance, and this rule of `test_my_function_name` wouldn't apply anyway. I haven't really caught up on the context here so it's not a strong opinion, but I generally find it more appealing when the name of the test makes sense semantically.

Ideally, we shouldn't have to write tests for private interfaces. 🙁

Reply:
It's reasonable for me and I removed underscores.
@HideakiImamura @hvy @toshihikoyanase Thank you for the bunch of discussions.